src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java

```diff
@@ -0,0 +1,584 @@
+                                    if (error.errorType.equals(NewGTFSErrorType.DUPLICATE_ID)) {
+                                        skipRecord = true;
+                                    }
+                                }
+                                break;
+                        }
+                        String valueToWrite = val;
+                        if (field.isForeignReference()) {
+                            // If the field is a foreign reference, check to see whether the reference has been
+                            // remapped due to a conflicting ID from another feed (e.g., calendar#service_id).
+                            String key = String.join(":", field.referenceTable.name, idScope, val);
+                            if (mergeFeedsResult.remappedIds.containsKey(key)) {
+                                mergeFeedsResult.remappedReferences++;
+                                // If the value has been remapped update the value to write.
+                                valueToWrite = mergeFeedsResult.remappedIds.get(key);
+                            }
+                            // If the current foreign ref points to another record that has been skipped, skip this
+                            // record and add its primary key to the list of skipped IDs (so that other references can
+                            // be properly omitted).
+                            if (mergeFeedsResult.skippedIds.contains(key)) {
+                                String skippedKey = String.join(":", table.name, idScope, keyValue);
+                                if (orderField != null) {
+                                    skippedKey = String.join(":",
+                                                             skippedKey,
+                                                             csvReader.get(getFieldIndex(fieldsFoundInZip, orderField))
+                                    );
+                                }
+                                mergeFeedsResult.skippedIds.add(skippedKey);
+                                skipRecord = true;
+                                continue;
+                            }
+                        }
+                        if (this.mergeType.equals(REGIONAL) && isKeyField && !val.isEmpty()) {
+                            // For regional merge, if field is a GTFS identifier (e.g., route_id, stop_id, etc.),
+                            // add scoped prefix.
+                            valueToWrite = String.join(":", idScope, val);
+                        }
+
+                        rowValues[specFieldIndex] = valueToWrite;
+                    }
+                    if (table.name.equals("routes") && rowValues[0].contains("BART")) {
+                        LOG.info("Route record: {}", rowValues[0]);
+                    }
+                    // Do not write rows that are designated to be skipped.
+                    if (skipRecord && this.mergeType.equals(MTC)) {
```

Comment:
oh snap, this kind of ties it all together for me here. So records are only skipped if the MTC merge type is enabled, otherwise the scoped IDs are used and the records are written. Is that right?

Reply:
Yes, that is correct.